include/TNetwork.h

```diff
@@ -27,6 +27,14 @@
 
 struct TConnection;
 
+class WatchingConnecting{
+public:
+    bool IsConnectionAllowed(const std::string& clientAddress);
+private:
+    void BlockIP(const std::string& clientAddress);
+    bool IsIPBlocked(const std::string& clientAddress);
+};
+
```

Comment:
Please rename to something like `RateLimiter` and move it to its own file (`RateLimiter.cpp` and `RateLimiter.h`, for example).